### DISCUSSION TITLE
New protocol implementation

### DISCUSSION THREAD

**Discussion by Lixlechat** [BishopFox/sliver#1148] [Q&A] [answered]
Hi everyone.

I wanted to add a new protocol (ModBusTCP/IP) . Can anyone help me to find where i need add code for create a the new protocol on Sliver?

=)

**Comment by moloch--** [comment, marked as answer]
We plan to make it easier to implement new C2 protocols in the future, right now it's not simple:

- Server-side code: https://github.com/BishopFox/sliver/tree/master/server/c2
- Implant-side code: https://github.com/BishopFox/sliver/tree/master/implant/sliver/transports

**Reply by Lixlechat** [reply]
i will try, thanks a lot!
